numpy/testing/tests/test_utils.py

```diff
@@ -1081,6 +1081,12 @@ def test_simple(self):
 
         assert_raises(AssertionError,
                       lambda: assert_string_equal("foo", "hello"))
+        
+    def test_regex(self):
+        assert_string_equal("a+*b", "a+*b")
```

Comment:
I wonder if it is worth using a parametrized test (from pytest) to probe a few more of the regular expression special characters used in Python. On the other hand, if I'm not mistaken, there may not be much drive for this kind of support in `numpy.testing` moving forward with increased adoption of pytest.

Reply:
> there may not be much drive for this kind of support in numpy.testing

How so? Does pytest offer an equivalent for the function being tested?

Reply:
Probably OK to skip complete coverage here unless there is a return to regular expressions.

@tylerjereddy Now that you are working a BIDS, do you need commit privileges?

Reply:
I guess that's up to the team, but I'm happy to continue without for now. I don't think `pytest` has a replacement functionality for what NumPy does here.